src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -106,7 +107,15 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
         {
             try
             {
-                result = _monarch.ProposeCommandline(args);
+                // MSFT:38542548 _We believe_ that this is the source of the
+                // crash here. After we get the result, stash it's values into a
+                // local copy, so that we can check them later. If the Monarch
+                // dies between now and the inspection of
+                // `result.ShouldCreateWindow` below, we don't want to explode
+                // (since _proposeToMonarch is not try/caught).
+                Remoting::ProposeCommandlineResult outOfProcResult = _monarch.ProposeCommandline(args);
+                result = winrt::make<implementation::ProposeCommandlineResult>(outOfProcResult);

```

Comment:
Copying the `outOfProcResult` still involves the exact same call to `ShouldCreateWindow` which might fail, right? So isn't the underlying bug, that we didn't protect the call to `ShouldCreateWindow` down below with a try/catch?

If I'm right, I don't like this fix _that_ much because it hides a potential source of exceptions inside a constructor call, which is a bit "opaque".

IMO we should consider moving this block from down below:
```cpp
_shouldCreateWindow = result.ShouldCreateWindow();
if (result.Id())
{
    givenID = result.Id().Value();
}
givenName = result.WindowName();
```

...up here, which would fix the bug the exact same way.

Reply:
Ah, unless of course you want to prevent those 3 variables from being in an inconsistent state.
I'd still prefer a solution with variables, but given that I'll approve the PR.